static/src/js/components/TemplateForm/TemplateForm.jsx

```diff
@@ -0,0 +1,331 @@
+
+  const fields = {
+    AnyOfField: () => null,
+    BoundingRectangle: BoundingRectangleField,
+    keywordPicker: KeywordPicker,
+    layout: GridLayout,
+    OneOfField,
+    streetAddresses: StreetAddressField,
+    TitleField: CustomTitleField
+  }
+
+  const widgets = {
+    CheckboxWidget: CustomRadioWidget,
+    CountrySelectWidget: CustomCountrySelectWidget,
+    DateTimeWidget: CustomDateTimeWidget,
+    RadioWidget: CustomRadioWidget,
+    SelectWidget: CustomSelectWidget,
+    TextareaWidget: CustomTextareaWidget,
+    TextWidget: CustomTextWidget
+  }
+  const templates = {
+    ArrayFieldTemplate: CustomArrayFieldTemplate,
+    FieldTemplate: CustomFieldTemplate,
+    TitleFieldTemplate: CustomTitleFieldTemplate
+  }
+  const { ummMetadata = {} } = draft || {}
+
+  const [firstFormSection] = collectionsTemplateConfiguration
+  const firstSectionName = kebabCase(firstFormSection.displayName)
+  const currentSection = sectionName || firstSectionName
+
+  // Limit the schema to only the fields present in the displayed form section
+  const formSchema = getFormSchema({
+    fullSchema: ummCTemplateSchema,
+    formConfigurations: collectionsTemplateConfiguration,
+    formName: currentSection
+  })
+
+  useEffect(() => {
+    // If fieldName was pulled from the URL, set it to the focusField
+    setFocusField(fieldName)
+
+    // If a fieldName was pulled from the URL, then remove it from the URL. This will happen after the field is focused.
+    if (fieldName && sectionName) navigate(`/templates/collections/${id}/${sectionName}`, { replace: true })
+  }, [fieldName])
+
+  // Fetching collection template if ID is present and draft is not loaded
+  useEffect(() => {
+    const fetchTemplate = async () => {
+      const { response, error: fetchTemplateError } = await getTemplate(providerId, token, id)
+
+      if (response) {
+        delete response.pathParameters
+
+        setDraft({
+          ummMetadata: { ...response }
+        })
+      } else { setErrors(fetchTemplateError) }
+
+      setLoading(false)
+    }
+
+    if (id !== 'new' && !draft) {
+      setLoading(true)
+      fetchTemplate()
+    }
+  }, [id])
+
+  const handleSave = async (type) => {
+    setSaveLoading(true)
+    let savedId = null
+    if (id === 'new') {
+      const response = await createTemplate(providerId, token, ummMetadata)
+
+      savedId = response
+      setSaveLoading(false)
+    } else {
+      const response = await updateTemplate(providerId, token, ummMetadata, id)
+
+      if (response.ok) {
+        addNotification({
+          message: 'Template saved successfully',
+          variant: 'success'
+        })
+      } else {
+        addNotification({
+          message: 'Error saving template',
+          variant: 'danger'
+        })
+
+        errorLogger('Error saving template', 'TemplateForm: updateTemplate')
+      }
+
+      setSaveLoading(false)
+    }
+
+    if (type === saveTypes.save) {
```

Comment:
Create a hook to call save and use that for TemplateForm and MetadataForm

Reply:
During PR review, an idea was brought up to create the save and naviagte into a hook or a utils so both TemplateForm and MetadataForm can use that to naviagte. After looking at both file, converting this into it's own helper funtion will make the function complicated beacuse the routes and onClick action are not similar for both components.